### PR TITLE
Add support for Billing CloudWatch metrics and alarms

### PR DESCRIPTION
**Pull request by catmeme**
Usage:

```
const estimatedCharges = awsx.billing.metrics.estimatedCharges({
  currency: "USD",
  period: 21600,
  statistic: "Maximum",
});

estimatedCharges.createAlarm("billing.estimatedCharges", {
  alarmActions: [topicAlerts],
  evaluationPeriods: 1,
  okActions: [topicAlerts],
  threshold: 1000,
  treatMissingData: "breaching",
});

const estimatedChargesEc2 = awsx.billing.metrics.estimatedCharges({
  currency: "USD",
  period: 21600,
  serviceName: "AmazonEC2",
  statistic: "Maximum",
});

estimatedChargesEc2.createAlarm("billing.AmazonEC2.estimatedCharges", {
  alarmActions: [topicAlerts],
  evaluationPeriods: 1,
  okActions: [topicAlerts],
  threshold: 100,
  treatMissingData: "breaching",
});
```

Adds https://github.com/pulumi/pulumi-awsx/issues/658

### CODE REVIEW

File: nodejs/awsx/billing/metrics.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    /**
+     * EstimatedCharges
+     *
+     * Units: currency
+     * Valid CloudWatch statistics: Average, Maximum (recommended), Minimum
+     */
+    export function estimatedCharges(change?: BillingMetricChange) {
+        return metric("EstimatedCharges", { statistic: "Maximum", currency: "USD", ...change });
```
Comment (stack72) [inline nodejs/awsx/billing/metrics.ts:77]: the question for me here is can "Statistic" and "currency" be anything other than these values above?

Comment (catmeme) [reply nodejs/awsx/billing/metrics.ts:77]: Statistic looks like it accepts the usual values, however, I was unable to find a list of supported currencies, and the only thing I've found is "USD".

**Review by stack72 — APPROVED** [review]
Thanks so much for this @catmeme :)

**Pull request merged**